### ISSUE TITLE
Using the NoPrivateNetworkHttpClient results in odd error-message without IP

### ISSUE THREAD

**Issue by heiglandreas** [symfony/symfony#59021]
### Symfony version(s) affected

7.1.9

### Description

We updated recently from http-client 7.1.8 to 7.1.9 and suddenly got a lot of errors like

```
IP "" is blocked for "https://xyz.example.com".
```

All triggered in [vendor/symfony/http-client/NoPrivateNetworkHttpClient.php:249](https://github.com/symfony/symfony/blob/136bde3a52ff650dd6e47c53a3dad0765837b437/src/Symfony/Component/HttpClient/NoPrivateNetworkHttpClient.php#L249)

The stack trace ends in

```
vendor/symfony/http-client/NoPrivateNetworkHttpClient.php:103
vendor/symfony/http-client/Response/AsyncContext.php:165
vendor/symfony/http-client/Response/AsyncContext.php:165
vendor/symfony/http-client/Response/CurlResponse.php:131
```

From what I understand it looks like the field `primary_ip` is empty and therefore considered not a valid IP-address at all (which makes sense). 

As this worked in version 7.1.8 (to which we reverted and the errors stopped immediately) it looks like something broke in the latest version.

### How to reproduce

See https://gitlab.com/heiglandreas/symfony-http-client-issue-59021

**Comment by Seldaek** [comment]
@nicolas-grekas could be related to https://github.com/symfony/symfony/pull/59013 but not sure

**Comment by nicolas-grekas** [comment]
Definitely. example.com makes me think this might be related to some handling of [RFC6761](https://datatracker.ietf.org/doc/html/rfc6761)

**Comment by heiglandreas** [comment]
The problem seems to be related to requests that first encounter a redirect (301) and then run into an error (4xx).

Edit:

Looks like it also happens when the end-result is a 200...

**Comment by heiglandreas** [comment]
The `example.com` was the redacted domain. Unless I have something reproducible I can't publish customer data publicly...

**Comment by heiglandreas** [comment]
I have a reproducible case available at https://gitlab.com/heiglandreas/symfony-http-client-issue-59021

The README contains the error-message including the `IP "" is blocked for "..."` message.

**Comment by nicolas-grekas** [comment]
Thanks, fix at #59023
